boot/bootutil/src/loader.c

```diff
@@ -630,7 +633,7 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
 #endif
 
     if (!boot_is_header_valid(hdr, fap) || boot_image_check(state, hdr, fap, bs)) {
-        if ((slot != BOOT_PRIMARY_SLOT) || IS_IN_XIP_MODE()) {
+        if ((slot != BOOT_PRIMARY_SLOT) || ARE_SLOTS_EQUIVALENT()) {
```

Comment:
`ARE_SLOTS_EQUIVALENT()` is introduced only for this one usage - it is less descriptive than direct mention of mcuboot `modes`
```suggestion
  if ((slot != BOOT_PRIMARY_SLOT) || defined(MCUBOOT_DIRECT_XIP) || defined(MCUBOOT_RAM_LOADING)){
```

Eventually change `ARE_SLOTS_EQUIVALENT()` -> `IS_IN_DIRECT_MODE()`

Reply:
^^ with commnet what this mean: either direct XIP or ram load

Reply:
ack

Reply:
It seems that the `defined` preprocessor directive cannot be used in a normal `if` statement. I got compilation error. I think this is the reason why the `IS_IN_DIRECT_MODE` macro was  introduced. @davidvincze suggested to rename the macro, because in case of `RAM_LOADING` the image is not XIP and the term `ARE_SLOTS_EQUIVALENT` expresses more the similarity between the two mode.

Reply:
I see. then keep as is.